test/functional/test_framework/wallet.py

```diff
@@ -301,9 +307,10 @@ def getnewdestination(address_type='bech32'):
     elif address_type == 'bech32':
         scriptpubkey = key_to_p2wpkh_script(pubkey)
         address = key_to_p2wpkh(pubkey)
-    # TODO: also support bech32m (need to generate x-only-pubkey)
-    else:
-        assert False
+    elif address_type == 'bech32m':
+        pubkey = taproot_construct(compute_xonly_pubkey(key.get_bytes())[0]).output_pubkey
+        scriptpubkey = output_key_to_p2tr_script(pubkey)
+        address = output_key_to_p2tr(pubkey)
```

Comment:
you might want to keep the `else: assert False` in case anyone passes an unknown string in (e.g. typo)

Reply:
Good idea, thanks. Done.